src/FreeAge/client/object.cpp

```diff
@@ -43,6 +43,14 @@ const Texture* ClientObject::GetIconTexture() const {
   return nullptr;
 }
 
+void ClientObject::GarrisonUnit(u32 unitId) { 
+  garrisonedUnits.emplace_back(unitId);
+}
+
+void ClientObject::UngarrisonUnit(u32 unitId) {
+  garrisonedUnits.erase(std::remove(garrisonedUnits.begin(), garrisonedUnits.end(), unitId), garrisonedUnits.end());
```

Comment:
Technical: Is this how you write a remove?

Reply:
I am not really familiar with what the standard library has to offer for this. Thus I would have written it as follows:
```c++
bool found = false;
for (usize i = 0, size = garrisonedUnits.size(); i < size; ++ i) {
  if (garrisonedUnits[i] == unitId) {
    garrisonedUnits.erase(garrisonedUnits.begin() + i);
    found = true;
    break;
  }
}
if (!found) {
  LOG(ERROR) << "Did not find unit ID to ungarrison in garrisonedUnits: " << unitId;
}
```
This should have the advantages that it only searches up to the searched-for ID and then aborts (assuming that IDs never occur twice in the list), and that it outputs an error if the ID is not actually in there. Possible drawback: If the ID is actually in there twice due to a bug, it only gets removed once.

*Personal opinion: The construct with `remove` and then `erase` involving all those iterators looks very un-intuitive to me. However, this does seem to be the recommended way, judging from https://en.cppreference.com/w/cpp/algorithm/remove*

Reply:
I am also not really familiar with what the standard library has to offer, so I will replace it for the code you wrote which more understandable.

I was looking for something like `vector.remove(unitId)` but I guess std thinks its not a very common operation to implement

Reply:
> I was looking for something like vector.remove(unitId) but I guess std thinks its not a very common operation to implement

Maybe we can implement something like that ourselves:

```c++
template <typename T>
bool remove_unique(vector<T>& container, T element) {
  for (usize i = 0, size = container.size(); i < size; ++ i) {
    if (container[i] == element) {
      container.erase(container.begin() + i);
      return true;
    }
  }
  return false;
}
```

Then the code becomes:

```c++
if (!remove_unique(garrisonedUnits, unitId)) {
  LOG(ERROR) << "Did not find unit ID to ungarrison in garrisonedUnits: " << unitId;
}
```

Reply:
I will leave the template thing to you ....